### PR TITLE
Fix API-gateway release pipeline

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Fix api-gateway release pipeline
- Use tag from PULL_BASE_REF
- Add remote for goreleaser

### CODE REVIEW

**Comment by barchw** [comment]
@kyma-project/prow could you please review?

File: prow/jobs/api-gateway/api-gateway-release.yaml [hunk 1]
```diff
@@ -112,11 +108,12 @@ postsubmits: # runs on main
...
         containers:
-          - image: "eu.gcr.io/kyma-project/external/goreleaser:v1.11.5"
+          - image: "eu.gcr.io/kyma-project/external/goreleaser/goreleaser:v1.11.5"
             command:
-              - "goreleaser"
+              - "/bin/sh"
             args:
-              - "release"
+              - "-c"
+              - "git remote add origin https://github.com/kyma-project/api-gateway && goreleaser release"
```
Comment (Ressetkk) [inline prow/jobs/api-gateway/api-gateway-release.yaml:116]: why did you have to add a remote here? This should be self-sufficient in most of the scenarios. Some explanation in the PR description of underlying problem would be nice.

Comment (barchw) [reply prow/jobs/api-gateway/api-gateway-release.yaml:116]: I was testing the pipeline execution on a test PR and got the following error: https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/barchw_test_of_prowjob_rel-api-gateway-goreleaser/1608458819308883968
After adding the origin ref manually the error is gone: https://status.build.kyma-project.io/view/gs/kyma-prow-logs/logs/barchw_test_of_prowjob_rel-api-gateway-goreleaser/1608461889103204352

Comment (Ressetkk) [reply prow/jobs/api-gateway/api-gateway-release.yaml:116]: Ah, ok. Let's leave it as is then.

**Review by Ressetkk — DISMISSED** [review]

**Pull request merged**